### PR TITLE
add custom deployment provider support

### PR DESCRIPTION
**Pull request by HomelessDinosaur**

### CODE REVIEW

File: pkg/provider/remote/custom.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+// TODO: Move this into remote provider logic
+// A provider should be able to produce it's own template for a valid stack specification as stack specifications may change over time
```
Comment (jyecusch) [inline pkg/provider/remote/custom.go:28]: Can this be done before the PR is merged?

File: pkg/provider/remote/custom.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+type customProvider struct {
+	*nitricDeployment
+}
+
+// FIXME: Prompting to create a new stack state and memory and persisting to a file should be two separte functions
+func (g *customProvider) AskAndSave() error {
```
Comment (jyecusch) [inline pkg/provider/remote/custom.go:34]: This too?

Comment (HomelessDinosaur) [reply pkg/provider/remote/custom.go:34]: Not yet, remote providers can't make templates yet cause there's no API for it. Will need to create a template and prompt API that the CLI can understand.

Comment (HomelessDinosaur) [reply pkg/provider/remote/custom.go:34]: Same with above ^^

File: pkg/provider/remote/custom.go (hunk 1 above)
Comment (tjholm) [reply pkg/provider/remote/custom.go:28]: This needs some work on the contracts for the remote providers, Probably some kind of service we can add the the proto contracts that allows us to output a template, but would need to be readable by the CLI, for user prompts/token replacement.

**Review by davemooreuws — APPROVED** [review]

**Pull request merged**